### PR TITLE
Prevent calls to EC2 metadata service for AWS C++ SDK users including HttpRequestor Gem unless explictly enabled.

### PR DESCRIPTION
**Pull request by lmbr-pip**
AWS C++ SDK is used under the hood to provide most of the functionality for the HttpRequestor Gem, as it offers a multi-platform http client that works well with O3DE. 

Unfortunately it ties that to a ClientConfiguration that insists on knowing the region and potentially looking up credentials to sign requests. This means it can make unexpected calls to the EC2 Metadata service, which wastes network traffic, generates cryptic log spam and will always fail on non EC2 compute.

We can avoid this by setting the env var [`AWS_EC2_METADATA_DISABLED`](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html) which is used to control the behaviour: https://github.com/aws/aws-sdk-cpp/blob/2ac79af8c551aff94ef5d28c88bf09209b0b71c8/aws-cpp-sdk-core/source/client/ClientConfiguration.cpp 
* Most users will never run code on EC2 instances so this behaviour is useless.

Change is a short term fix which make defaults to this behaviour being off by default, with a cvar to turn it on. Set ownership at the AWSNativeSDK level to avoid any code fighting over setting environment variable. 

Will need a paired change in the AWS GameLift Gem to add to logic to control how to turn on this setting or add set/reg expansion so AWSNativeSDK will work correctly when hosted on GameLift (which is EC2 compute backed). Will create issue to cover that work as it requires thought and potentially to cover other use cases related to setting the environment variables based config options, which may require a change to the wrapper to control config directly.

## Testing Notes
Expanded the HttpRequestor "automated test" to cover checking for reach out to EC2. Looked for other unit test places but as this is mostly setting an environment variable, testing is most useful at integration test level.

##Tests so far (Windows only)
* Confirmed calls to EC2 metadata services with unaltered code
* Turned on trace logging
* Confirmed flipping variable on worked as expected - no calls to EC2 metadata service
* Confirmed automated test now passes with trace logging on as well as warn

## Work still todo
[] Test on Linux

Signed-off-by: Pip Potter <61438964+lmbr-pip@users.noreply.github.com>

### CODE REVIEW

File: Code/Tools/AWSNativeSDKInit/source/AWSNativeSDKInit.cpp [hunk 1]
```diff
@@ -79,6 +81,9 @@ namespace AWSNativeSDKInit
         Platform::CustomizeSDKOptions(m_awsSDKOptions);
         Aws::InitAPI(m_awsSDKOptions);
 
+        //! Disable calls to EC2Metadata service by default
+        AZ::Utils::SetEnv("AWS_EC2_METADATA_DISABLED", "True", true);
```
Comment (petrocket) [inline Code/Tools/AWSNativeSDKInit/source/AWSNativeSDKInit.cpp:85]: Another option would be to add a setting in the settings registry to control this value.  That way you could control it without compiling.

Comment (lmbr-pip) [reply Code/Tools/AWSNativeSDKInit/source/AWSNativeSDKInit.cpp:99]: Added a cvar to make toggling this on/off easier. May need to find new owner of fix, if we need to support setreg as well.

File: Code/Tools/AWSNativeSDKInit/source/AWSNativeSDKInit.cpp [hunk 2]
```diff
@@ -21,6 +25,17 @@
 
 namespace AWSNativeSDKInit
 {
+    // Prevent/allow the use of the Amazon EC2 instance metadata service (IMDS). 
+    // AWS C++ SDK can reach out to EC2 metadata service for region, config or credentials, but unless code
+    // is running on EC2 compute such calls will fail and waste network resources.
+    void OnBlockEC2MetadataCalls(const bool& setMode)
+    {
+        AZ::Utils::SetEnv("AWS_EC2_METADATA_DISABLED", (setMode ? "True" : "False"), true);
```
Comment (junbo75) [inline Code/Tools/AWSNativeSDKInit/source/AWSNativeSDKInit.cpp:33]: Minor: What if users already set the AWS_EC2_METADATA_DISABLED environment variable on their machine? May worth checking whether the variable has already been set before override it.

File: Gems/HttpRequestor/Code/Tests/HttpRequestorTest.cpp [hunk 3]
```diff
@@ -8,15 +8,67 @@
...
+        }
+        if (!AZ::Interface<AZ::Debug::IEventLogger>::Get())
+        {
+            m_eventLogger.reset(new AZ::Debug::LocalFileEventLogger);
+        }
+    }
+
+    void TearDown() override
+    {
+        BusDisconnect();
```
[the remark this answers is not in the document]
Comment (lmbr-pip) [reply Gems/HttpRequestor/Code/Tests/HttpRequestorTest.cpp:53]: Fixed

**Comment by lmbr-pip** [comment]
Am going to ask for this code to be accepted as is, because:

* There is no current way to check to see if environmental variable is set in advance from o3DE. We only have platform native setenv (https://github.com/o3de/o3de/issues/10033 - feature request for GetEnv equivalent)
* Users only need to allow calls to EC2 metadata service if they are running on EC2 compute (a rare event for O3DE code - mostly O3DE servers on EC2), for the 90% use case, turning off EC2 Metadata service calls is safe and the right thing todo even for AWS Gems. 
* I don't want to tie AWSNativeSDKInit to settings registry because a) its unnecessary coupling and b) we are planning to re-work AWSNativeSDKInit code as part of this RFC: https://github.com/o3de/sig-network/issues/64. The right thing todo is to provide some form of Config options that can be passed in, calling code can then use settings registry checks to inform what the config should be.

Will need to make a small docs update to cover the new CVar and option and will also revisit once full GameLift automation is delivered to the project, where you may need to pull server region/creds from EC2 metadata service.

**Review by AMZN-alexpete — APPROVED** [review]

**Review by AMZN-puvvadar — APPROVED** [review]

**Review by allisaurus — APPROVED** [review]

**Comment by lmbr-pip** [comment]
Have somehow destroyed linkage to upstream, will recreate in new PR.

**Comment by lumberyard-employee-dm** [comment]
This can be fixed, by instead of trying to merge latest development, rebasing on `upstream/development`

**Pull request closed without merging**